controlplane/src/core/repositories/NamespaceRepository.ts

```diff
@@ -0,0 +1,73 @@
+import { and, eq } from 'drizzle-orm';
+import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
+import { EnumStatusCode } from '@wundergraph/cosmo-connect/dist/common/common_pb';
+import * as schema from '../../db/schema.js';
+import { PublicError } from '../errors/errors.js';
+
+export const DefaultNamespace = 'default';
+
+export class NamespaceRepository {
+  constructor(
+    private db: PostgresJsDatabase<typeof schema>,
+    private organizationId: string,
+  ) {}
+
+  public byName(name: string) {
+    return this.db.query.namespaces.findFirst({
+      where: and(eq(schema.namespaces.organizationId, this.organizationId), eq(schema.namespaces.name, name)),
+    });
+  }
+
+  public async byTargetId(id: string) {
+    const res = await this.db.query.namespaces.findMany({
+      where: and(eq(schema.namespaces.organizationId, this.organizationId)),
+      with: {
+        targets: {
+          columns: {
+            id: true,
+          },
+        },
+      },
+    });
+
+    return res.find((r) => r.targets.find((t) => t.id === id));
+  }
+
+  public async create(data: { name: string; createdBy: string }) {
+    const ns = await this.db
+      .insert(schema.namespaces)
+      .values({
+        name: data.name,
+        organizationId: this.organizationId,
+        createdBy: data.createdBy,
+      })
+      .returning();
+
+    if (ns.length === 0) {
+      throw new PublicError(EnumStatusCode.ERR, 'Could not create namespace');
```

Comment:
Repositories should not throw public errors, there are low level constructs. That should be done on the service level.

Reply:
Fixed